From 9a2dc2d1e8dedae64472437486088b58999ab7dd Mon Sep 17 00:00:00 2001
From: Sailaja <sbandarupalli@habana.ai>
Date: Wed, 30 Nov 2022 08:44:21 +0200
Subject: [PATCH] [SW-0] gaudi2: correct efuse bit offsets

arcbootrom writes the value of efuse offset 0x3A at a different
location in SPSRAM (192). So if SPSRAM is read to get efuse value
then SECURITY_EFUSE_SYS_CFG_OFFSET should be used instead of
efuse offset 0x3A

Change-Id: I5137fab12ff105ee95087e14d0d89661b3cbb4d0
Signed-off-by: Sailaja <sbandarupalli@habana.ai>
(cherry picked from commit af7965c6b4e2bd28d8d9dae32e58f52c27ad748e)
---

diff --git a/arch/gaudi2/arc/common.c b/arch/gaudi2/arc/common.c
index eb7e7e2..2914a05 100644
--- a/arch/gaudi2/arc/common.c
+++ b/arch/gaudi2/arc/common.c
@@ -539,7 +539,7 @@
 
 bool pcie_skip_wa_snps(void)
 {
-	uint32_t config_fuse = sp_sram_readl(EFUSE_SYSCFG_OFFSET);
+	uint32_t config_fuse = sp_sram_readl(SECURITY_EFUSE_SYS_CFG_OFFSET);
 
 	if ((config_fuse & EFUSE_SYSCFG_WA_SNPS_CASE_01038891) == 0)
 		return false;
@@ -551,7 +551,7 @@
 {
 	uint8_t separate_reference = 0;
 	uint8_t ssc_en = 0, srns = 0;
-	u32 config_fuse = sp_sram_readl(EFUSE_SYSCFG_OFFSET);
+	u32 config_fuse = sp_sram_readl(SECURITY_EFUSE_SYS_CFG_OFFSET);
 
 	separate_reference = get_sris_mode_from_strap();
 
@@ -643,7 +643,7 @@
 {
 	uint32_t boot_strap, link_ctrl;
 	uint8_t ltssm_en;
-	u32 config_fuse = sp_sram_readl(EFUSE_SYSCFG_OFFSET);
+	u32 config_fuse = sp_sram_readl(SECURITY_EFUSE_SYS_CFG_OFFSET);
 
 	pcie_loop_while_prstn(PCIE_PRSTN_ASSERT);
 	ltssm_en = regmap_readl(mmPCIE_AUX_LTSSM_EN) &
